client/server/pages/index.js

```diff
@@ -681,23 +680,9 @@ module.exports = function() {
 		next();
 	} );
 
-	if ( jetpackCloudEnvs.includes( calypsoEnv ) ) {
-		JETPACK_CLOUD_SECTION_DEFINITION.paths.forEach( sectionPath =>
-			handleSectionPath( JETPACK_CLOUD_SECTION_DEFINITION, sectionPath, 'entry-jetpack-cloud' )
-		);
-
-		// catchall to render 404 for all routes not whitelisted in client/sections
-		app.use( render404( 'entry-jetpack-cloud' ) );
-
-		// Error handling middleware for displaying the server error 500 page must be the very last middleware defined
-		app.use( renderServerError( 'entry-jetpack-cloud' ) );
-
-		return app;
-	}
-
 	// redirect homepage if the Reader is disabled
 	app.get( '/', function( request, response, next ) {
-		if ( ! config.isEnabled( 'reader' ) ) {
+		if ( ! config.isEnabled( 'reader' ) && ! jetpackCloudEnvs.includes( calypsoEnv ) ) {
```

Comment:
Could we use a feature flag instead of detecting the environment here? Last time we discussed this I thought we're trying to orient that kind of logic strongly around using feature flags. Also, that's what we do for the reader here already.

Reply:
Switched to use a feature flag.